src/wasm/wasm-type.cpp

```diff
@@ -601,17 +597,19 @@ bool Type::isSubType(Type left, Type right) {
     }
     // Various things are subtypes of eqref.
     auto leftHeap = left.getHeapType();
+    auto rightHeap = right.getHeapType();
     if ((leftHeap == HeapType::i31 || leftHeap.isArray() ||
          leftHeap.isStruct()) &&
-        right == Type::eqref) {
+        rightHeap == HeapType::eq &&
+        (!left.isNullable() || right.isNullable())) {
```

Comment:
Just noticed that the `right == Type::funcref` check below should probably be changed in the same way.

Reply:
Good catch, fixed.